common/beerocks/bcl/source/network/get_iface_stats.cpp

```diff
@@ -0,0 +1,243 @@
+{
+    bool result = false;
+
+    struct sockaddr_nl socket {
+    }; /* the remote (kernel space) side of the communication */
+    struct msghdr rtnl_msg {
+    }; /* generic msghdr struct for use with sendmsg */
+    struct iovec io {
+    }; /* IO vector for sendmsg */
+    struct nl_req_t req {
+    }; /* structure that describes the Netlink packet itself */
+
+    /**
+     * Netlink socket is ready for use, prepare and send request
+     */
+    socket.nl_family = AF_NETLINK; /* fill-in kernel address (destination of our message) */
+
+    req.hdr.nlmsg_len    = NLMSG_LENGTH(sizeof(struct rtgenmsg));
+    req.hdr.nlmsg_type   = RTM_GETLINK;
+    req.hdr.nlmsg_flags  = NLM_F_REQUEST | NLM_F_DUMP;
+    req.hdr.nlmsg_seq    = 1;
+    req.hdr.nlmsg_pid    = 0;
+    req.gen.rtgen_family = AF_PACKET; /*  no preferred AF, we will get *all* interfaces */
+
+    io.iov_base          = &req;
+    io.iov_len           = req.hdr.nlmsg_len;
+    rtnl_msg.msg_iov     = &io;
+    rtnl_msg.msg_iovlen  = 1;
+    rtnl_msg.msg_name    = &socket;
+    rtnl_msg.msg_namelen = sizeof(socket);
+
+    if (sendmsg(fd, static_cast<struct msghdr *>(&rtnl_msg), 0) < 0) {
+        LOG(ERROR) << "Unable to send message through Netlink socket: " << strerror(errno);
+    } else {
+        bool msg_done = false; /* flag to end loop parsing */
+
+        /**
+         * Parse reply until message is done
+         */
+        while (!msg_done) {
+            struct msghdr rtnl_reply {
+            }; /* generic msghdr structure for use with recvmsg */
+
+            /* a large buffer to receive lots of link information */
+            char reply[IFLIST_REPLY_BUFFER];
+
+            io.iov_base            = reply;
+            io.iov_len             = IFLIST_REPLY_BUFFER;
+            rtnl_reply.msg_iov     = &io;
+            rtnl_reply.msg_iovlen  = 1;
+            rtnl_reply.msg_name    = &socket;
+            rtnl_reply.msg_namelen = sizeof(socket);
+
+            /**
+             * Read as much data as fits in the receive buffer
+             */
+            int length = recvmsg(fd, &rtnl_reply, 0);
+            for (const struct nlmsghdr *msg_ptr = reinterpret_cast<const struct nlmsghdr *>(reply);
+                 NLMSG_OK(msg_ptr, length); msg_ptr = NLMSG_NEXT(msg_ptr, length)) {
+                switch (msg_ptr->nlmsg_type) {
+                case NLMSG_DONE:
+                    /**
+                     * This is the special meaning NLMSG_DONE message we asked for by using NLM_F_DUMP flag
+                     */
+                    msg_done = true;
+                    break;
+                case RTM_NEWLINK:
+                    /**
+                     * This is a RTM_NEWLINK message, which contains lots of information about a link
+                     */
+                    if (get_iface_stats(msg_ptr, iface_name, iface_stats)) {
+                        msg_done = true;
+                        result   = true;
```

Comment:
you can return from here with `return true`

Reply:
In this case, there is no unnecessary indentation, so I'll leave it as it is now